### ISSUE TITLE
No longer working

### ISSUE THREAD

**Issue by kalaws** [wallento/sungrow-websocket#7]
The script seems no longer to work since I updated to the latest SW for the WiNet-S module on my SH10RT. Anyone else having   problems? @damolp ?
```
        async with websockets.client.connect(
            f"ws://{self.host}:{self.port}/ws/home/overview"
        ) as websocket:
            await websocket.send(
                json.dumps(
                    {"lang": self.locale, "token": "", "service": "connect"}
                )
            )
            d: Result = json.loads(await websocket.recv())
            if d["result_code"] != 1 or d["result_msg"] != "success":
                return data
            token: str = d["result_data"]["token"]
            print(d)
            await websocket.send(
                json.dumps(
                    {
                        "lang": self.locale,
                        "token": token,
                        "service": "devicelist",
                        "type": "0",
                        "is_check_token": "0",
                    }
                )
            )
            d = json.loads(await websocket.recv())
            print(d)
```
returns
```
{'result_code': 1, 'result_msg': 'success', 'result_data': {'service': 'connect', 'token': '****redacted****', 'uid': 1, 'tips_disable': 0, 'virgin_flag': 0, 'isFirstLogin': 0, 'forceModifyPasswd': 0}}
{'result_code': 401, 'result_msg': 'I18N_COMMON_INTER_ABNORMAL', 'result_data': {'service': 'devicelist'}}
{"Body": {}}
```

**Comment by kalaws** [comment]
Potentially the problem is related to authentication. Compare the following (I’m not sure myself how to implement this):

[](https://github.com/bohdan-s/SungrowModbusWebClient/compare/main...caliph9k:SungrowModbusWebClient:main)

**Comment by mbaul** [comment]
I've also noticed this. I'm keen to hear if there is an update available and happy to help in testing.

**Comment by naturalnetworks** [comment]
I updated the firmware on my WiNET-S the other day and found this same issue so I've added SSL/TLS (Secure Websockets) and authentication - seems to work, haven't done any sort of thorough testing though. https://github.com/naturalnetworks/sungrow-websocket

**Comment by NullPointerExceptionError** [comment]
> I updated the firmware on my WiNET-S the other day and found this same issue so I've added SSL/TLS (Secure Websockets) and authentication - seems to work, haven't done any sort of thorough testing though. https://github.com/naturalnetworks/sungrow-websocket

@naturalnetworks Two days ago, the original version of wallento stopped working for me for exactly the same reason as the introduction of HTTPS. That's why I switched to your version of websocket (thanks for that). It worked great for a day. Now I only get timeout errors. When I access WiNet-S via a browser, it also takes a long time to load. Any idea what the problem could be?

**Comment by naturalnetworks** [comment]
@NullPointerExceptionError I'm having the same issue. I've tried a few different things to get it working however it seems to go okay for around a day but then gets progressively worse until it stops. The only way to get it going again is to restart the WiNET-S itself (I do so via the management interface or app). I'm out of ideas about what it could be.

**Comment by wallento** [comment]
Hi,Sorry, I will need to look into it now. Also had the update and can reproduce the problem. It seems the easy way I accessed it is gone, but I hope I can soon look into alternatives!Am 28.03.2025 um 05:29 schrieb Ben Johns ***@***.***>:﻿
  @NullPointerExceptionError I'm having the same issue. I've tried a few different things to get it working however it seems to go okay for around a day but then gets progressively worse until it stops. The only way to get it going again is to restart the WiNET-S itself (I do so via the management interface or app). I'm out of ideas about what it could be.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



naturalnetworks left a comment (wallento/sungrow-websocket#7)
@NullPointerExceptionError I'm having the same issue. I've tried a few different things to get it working however it seems to go okay for around a day but then gets progressively worse until it stops. The only way to get it going again is to restart the WiNET-S itself (I do so via the management interface or app). I'm out of ideas about what it could be.

**Comment by keremerkan** [comment]
Hi,

As I also encountered the degraded performance with the new firmware, I moved to [SungrowInverter with MODBUS](https://github.com/mvandersteen/SungrowInverter). So far it works perfectly and it outputs JSON (even though Sungrow's JSON format is terrible and needs some fixing)

If anyone would like to move to that, take these into consideration:

1. Enable MODBUS port 502 without SSL on your WiNet-S.
2. Make sure pymodbus and pycryptodomex are installed with the minimum versions stated in the pyproject.toml file (newer versions seem to cause errors)
3. Connect to port 502 on your device and make sure to set timeout to a few seconds (like 5 for example). If you don't set it, the script waits for the default 60 seconds even though it gathers all information.
4. Fix the JSON errors in the output. Convert single quotes to double quotes and fix TRUE/FALSE values.
5. Profit!

Thanks to @wallento for his great work, but it looks like it is now time to move on from websockets.

**Comment by keremerkan** [comment]
I don’t think it would be worth the effort. Moving to SungrowInverter was very straightforward, so wrapping it again may overcomplicate things. Sunsetting the project and redirecting users to SungrowInverter would be easier for you in my opinion.

**Comment by keremerkan** [comment]
Though now I remembered that it does not have a command line interface. Perhaps adding that would be beneficial.
